design-docs/improved-console.md

```diff
@@ -0,0 +1,422 @@
+ common Linux Terminals
+* Console shows idle workers under `--continuous` build
+
+## Story: Log messages can be configured to give more context
+With the removal of task execution logging in the main area, users need to get some additional
+context when warnings or errors occur. 
+
+### User Visible Changes
+Users see log level and build operation description for WARN and ERROR level 
+when rendering log events that have a severity and build operation specified.
+
+**Before**:
+```
+Note: Some input files use unchecked or unsafe operations.
+Note: Recompile with -Xlint:unchecked for details.
+Log message from build logic
+```
+
+**After**:
+```
+[:util:compileJava] Note: Some input files use unchecked or unsafe operations.
+[:util:compileJava] Note: Recompile with -Xlint:unchecked for details.
+Log message from build logic
+```
+
+### Implementation
+The concept of a source operation is added to `RenderableOutputEvent`. `LogEvent`s
+and `StyledTextOutputEvent` can be constructed with a String that generally 
+contains the `BuildOperation` description. 
+
+### Test Cases
+* Log messages without source operation configured renders same as usual
+
+### Open Issues
+* Should categories be leveraged instead?
+* Concept of log level or severity
+* Additions needed to `org.gradle.api.Logger` and `StyledTextOutput` for use in build scripts
+
+## Story: Allow user to include additional info in log messages
+Allow users to specify whether timestamp, log level, and/or category are logged with each message.
+
+Format is constant as "%dateTime% [%logLevel%] [%category%]"
+
+### Implementation
+Change logic of `StyledTextOutputBackedRenderer` to be configured with new Gradle properties
+
+```bash
+org.gradle.logger.showCategory=true                        # default=false
+org.gradle.logger.dateTimeFormat=yyyy-MM-dd HH:mm:ss:SSS Z # default=HH:mm:ss.SSS
+org.gradle.logger.showDateTime=true                        # default=false
+org.gradle.logger.showLogLevel=true                        # default=true
+```
+
+When rendering `RenderableOutputEvent`s, user configuration is taken into account.
+
+### Open issues
+* This should take source (build operation) into consideration
+
+## Story: Display overall build progress as a progress bar
+Render ProgressEvents with build completeness information as a progress bar.
+ 
+Intended to give the user a very fast way of telling whether the build will be finished soon or not.
+The work-in-progress (yellow) section shows how much of the build would be complete if all
+Operations displayed in the progressArea completed.
+
+### User-visible Changes
+We can visually represent complete and un-started tasks of build using colorized output:
+
+`#####green#####>###########black##########> 40% Building`
+
+On ANSI terminals, we can use empty spaces with different background colors for the "bars" and
+A red background shall be used for failed tasks. 
```

Comment:
Yes, individual slices would turn red. We know how many tasks are to be executed, from that we can calculate the size of any "slice". For example, suppose there are 7 tasks to be executed and one of them fails. That means ~14% of all tasks. We have a progress bar width of 50 chars, which means 7 characters wide. There would be a min-width of 1 character, even for 1/10000 tasks, of course.

We can make the whole thing red just like TeamCity does. That might be a better UX. I figured it'd be interesting for a user to know how "far" a build got before it failed... After thinking a bit about it I think I like your idea better.